File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> map;
        for(int i : arr) {
            map[i] += 1;
        }
        unordered_set<int> set;
        for(auto i : map) {
            set.insert(i.second);
        }
        return map.size() == set.size();
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        hmap = defaultdict(int)
        for i in arr :
            hmap[i] += 1
        hset = set()
        for i, j in hmap.items() :
            if j in hset : return False
            else : hset.add(j)
        return True